build-automation.js

```diff
@@ -0,0 +1,137 @@
+const util = require("util")
+
+const exec = util.promisify(require("child_process").exec);
+
+const https = require("https");
+
+const mapSeries = (arr) => {
+  const length = arr.length;
+  const results = new Array(length);
+
+  arr.reduce((chain, item, i) => {
+    return chain.then(() => item).then(val => results[i] = val);
+  }, Promise.resolve())
+  .then(() => results);
+}
+
+// a function that takes an URL as argument and makes a request to that URL
+// returning the response as a promise
+const request = (url) => {
+  return new Promise((resolve, reject) => {
+    https.get(url, async (res) => {
+      let body = '';
+
+      if(res.statusCode === 301 || res.statusCode === 302) {
+        resolve(await request(res.headers.location, resolve, reject));
+      }
+
+      res.on('data', (chunk) => {
+        body += chunk;
+      });
+      res.on('end', () => {
+        if (res.statusCode < 400) {
+          resolve(body);
+        } else {
+          reject(new Error(`Request failed: ${res.statusCode}`));
+        }
+      })
+    });
+  });
+};
+
+// a function that queries the Node.js release website for new versions,
+// compare the available ones with the ones we use in this repo
+// and returns whether we should update or not
+const checkIfThereAreNewVersions = async () => {
+  try {
+    const nodeWebsite = await request('https://nodejs.org/en/download/releases/');
+    const nodeWebsiteText = nodeWebsite.toString();
+
+    const { stdout: versionsOutput } = await exec(". ./functions.sh && get_versions", { shell: "bash" });
```

Comment:
I'd love to rewrite the scripts we have today into JS, but I guess just spawning them are fine for now.

Reply:
We can rewrite the scripts down the line. For now, I just think they went amazingly well with what I needed to do haha